mlflow/utils/_capture_transformers_modules.py

```diff
@@ -0,0 +1,92 @@
+"""
+This script should be executed in a fresh python interpreter process using `subprocess`.
+"""
+import importlib
+import json
+import sys
+import types
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.utils._capture_modules import _CaptureImportedModules, parse_args, store_imported_module
+
+
+class _CaptureImportedModulesForHF(_CaptureImportedModules):
+    """
+    A context manager to capture imported modules by temporarily applying a patch to
+    `builtins.__import__` and `importlib.import_module`.
+    Used for 'transformers' flavor only.
+    """
+
+    def __init__(self, module_to_throw):
+        super().__init__()
+        self.module_to_throw = module_to_throw
+
+    def _wrap_package(self, name):
+        if name == self.module_to_throw or name.startswith(f"{self.module_to_throw}."):
+            raise ImportError(f"Disabled package {name}")
+
+    def _record_imported_module(self, full_module_name):
+        self._wrap_package(full_module_name)
+        return super()._record_imported_module(full_module_name)
+
+    def __enter__(self):
+        import transformers
+
+        self.original_tf_available = transformers.utils.import_utils._tf_available
+        self.original_torch_available = transformers.utils.import_utils._torch_available
+        self.original_torch_fx_available = transformers.utils.import_utils._torch_fx_available
```

Comment:
Relying on private attributes is fragile. We should avoid it unless there is no other way.

Reply:
After putting this script into subprocess, environment variables actually works without the need to reloading all modules! I'll push the update :D